### PR TITLE
Feature/cpack_nsis

### PR DESCRIPTION
**Pull request by FIXMESTEVE**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,3 +37,41 @@ target_link_libraries(Overlay PRIVATE DearImgui DearImgui_Win32 DearImgui_DX9 Ga
...
+  CONFIGURATIONS Release
+  DESTINATION "."
+)
+
+# If NSIS is disabled at CMake configure step, the app will be packaged in a zip
+if(USE_NSIS)
+  set(CPACK_GENERATOR "NSIS")
+else(USE_NSIS)
+  set(CPACK_GENERATOR "ZIP")
+endif(USE_NSIS)
```
Comment (adanducci) [inline CMakeLists.txt]: Per the docs:
> If not specified, CPack will create a set of options following the naming pattern `CPACK_BINARY_<GENNAME>` (e.g. `CPACK_BINARY_NSIS`) allowing the user to enable/disable individual generators.

We could set `CPACK_BINARY_NSIS` `ON` by default in the CMake cache, and then let `cpack` do whatever it wants with the others- that way developers can opt into different ways of packaging things, but we have a sane release default.

File: CMakeLists.txt [hunk 2]
```diff
@@ -34,3 +37,41 @@ target_link_libraries(Overlay PRIVATE DearImgui DearImgui_Win32 DearImgui_DX9 Ga
...
+
+# TODO: Find a good variable to use instead of hardwriting "Launcher" (maybe ${EXE_TARGET_NAME})
+set(CPACK_PACKAGE_NAME "${PROJECT_NAME}")
+set(CPACK_PACKAGE_VENDOR "${PROJECT_NAME} Dev Team")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "${PROJECT_NAME} DESCRIPTION")
```
Comment (adanducci) [inline CMakeLists.txt]: Why not just use `${PROJECT_DESCRIPTION}`?

File: CMakeLists.txt [hunk 3]
```diff
@@ -34,3 +37,41 @@ target_link_libraries(Overlay PRIVATE DearImgui DearImgui_Win32 DearImgui_DX9 Ga
...
+set(CPACK_PACKAGE_VERSION "0.0.1")
+set(CPACK_PACKAGE_VERSION_MAJOR "0")
+set(CPACK_PACKAGE_VERSION_MINOR "0")
+set(CPACK_PACKAGE_VERSION_PATCH "1")
```
Comment (adanducci) [inline CMakeLists.txt]: The [docs suggest](https://cmake.org/cmake/help/v3.16/module/CPack.html#variable:CPACK_PACKAGE_VERSION_MAJOR) these are autofilled based on the `project()` command at the top of the file. Do we need these?

File: CMakeLists.txt [hunk 4]
```diff
@@ -34,3 +37,41 @@ target_link_libraries(Overlay PRIVATE DearImgui DearImgui_Win32 DearImgui_DX9 Ga
...
+set(CPACK_PACKAGE_INSTALL_DIRECTORY "${PROJECT_NAME}")
+set(CPACK_NSIS_MUI_ICON "${PROJECT_SOURCE_DIR}/resources/graphics/Launcher.ico")
+set(CPACK_NSIS_INSTALLED_ICON_NAME "${PROJECT_SOURCE_DIR}/resources/graphics/Launcher.ico")
+set(CPACK_NSIS_HELP_LINK "https://github.com/adanducci/GGPOPLUSR")
+set(CPACK_NSIS_URL_INFO_ABOUT "https://github.com/adanducci/GGPOPLUSR")
```
Comment (adanducci) [inline CMakeLists.txt]: These can probably just be the [project homepage URL](https://cmake.org/cmake/help/v3.16/variable/CMAKE_PROJECT_HOMEPAGE_URL.html#cmake-project-homepage-url).

File: CMakeLists.txt [hunk 5]
```diff
@@ -34,3 +37,41 @@ target_link_libraries(Overlay PRIVATE DearImgui DearImgui_Win32 DearImgui_DX9 Ga
...
+set(CPACK_NSIS_EXECUTABLES_DIRECTORY ".")
+set(CPACK_NSIS_MODIFY_PATH OFF)
+set(CPACK_NSIS_MENU_LINKS "Launcher.exe" "Start ${PROJECT_NAME}")
```
Comment (adanducci) [inline CMakeLists.txt]: +1 to your comment earlier. We should be able to get the executable name by grabbing the [`OUTPUT_NAME` property](https://cmake.org/cmake/help/v3.16/prop_tgt/OUTPUT_NAME.html) off the `Launcher` target via [`get_target_properties`](https://cmake.org/cmake/help/v3.16/command/get_target_property.html#command:get_target_property).

**Comment by FIXMESTEVE** [comment]
> If there's only one thing you can tweak before merging this, I would suggest updating the `README` to include packaging instructions. Do that, and I'll approve the merge.
> 
> For my own edification since I'm not at a Windows machine that can build right now, does the installer include `GGPO.dll`? And if it does, where does the installer place `GGPO.dll`?

I have updated the branch to implement your requests.
In this PR `GGPO.dll` is not included. Including `GGPO.dll` and `vcredist` is my next step.

**Pull request merged**